reactor-test/src/main/java/reactor/test/DefaultStepVerifierBuilder.java

```diff
@@ -423,15 +423,93 @@ else if (!predicate.test(signal.getSubscription())) {
 		return this;
 	}
 
-	final DefaultVerifySubscriber<T> build() {
-		return new DefaultVerifySubscriber<>(this);
+	final DefaultStepVerifier<T> build() {
+		return new DefaultStepVerifier<>(this);
 	}
 
 	@SuppressWarnings("unused")
 	interface Event<T> {
 
 	}
 
+	final static class DefaultStepVerifier<T> implements StepVerifier {
+
+		private final DefaultStepVerifierBuilder<T> parent;
+		private final int requestedFusionMode;
+		private final int expectedFusionMode;
+
+		DefaultStepVerifier(DefaultStepVerifierBuilder<T> parent) {
+			this.parent = parent;
+			this.requestedFusionMode = parent.requestedFusionMode;
+			this.expectedFusionMode = parent.expectedFusionMode == -1 ? parent.requestedFusionMode : parent.expectedFusionMode;
+		}
+
+		@Override
+		public Duration verify() {
+			return verify(Duration.ZERO);
+		}
+
+		@Override
+		public Duration verify(Duration duration) {
+			Objects.requireNonNull(duration, "duration");
+			if (parent.sourceSupplier != null) {
+				VirtualTimeScheduler vts = null;
```

Comment:
@smaldini I think it could make sense to consider the vts even in the case where sourceSupplier is null (probably because one wants to deal with the `Subscriber` directly, but still passed a specific `VirtualTimeScheduler` that he/she wants to manipulate by hand)